R/read.R

```diff
@@ -99,22 +99,24 @@ readH5AD <- function(file, X_name = NULL, use_hdf5 = FALSE,
                 ...)
 }
 
+#' @importFrom utils compareVersion
 #' @importFrom S4Vectors I DataFrame wmsg
-#' @importFrom SummarizedExperiment rowData colData rowData<- colData<-
+#' @importFrom SummarizedExperiment assays assays<- rowData colData rowData<- colData<-
 #' @importFrom SingleCellExperiment SingleCellExperiment reducedDims<- colPairs<- rowPairs<-
-.native_reader <- function(file, backed = FALSE, verbose = FALSE) {
+.native_reader <- function(file, backed = FALSE, verbose = FALSE, version = NULL) {
     .ui_info("Using the {.field R} reader")
     .ui_step("Reading {.file {file}}", spinner = TRUE)
 
+    version <- match.arg(version, .AnnDataVersions)
+
```

Comment:
I'm not sure if this is the right thing to do. The current `.AnnDataVersions` records the versions there are **{basilisk}** environments for but that might not match up with all the possible Python versions. It probably depends a bit what this is used for.

Reply:
Agreed. I changed it now to:
```
if (is.null(version)) {
    version <- .AnnDataVersions[1]
}
```
so when it's null it'll match the default python version, but otherwise isn't constrained.